### PR TITLE
Check for play, playbook and image path sanity

### PR DESCRIPTION
**Pull request by jpodivin**

### CODE REVIEW

File: controllers/openstack_ansibleee_controller.go [hunk 1]
```diff
@@ -345,12 +355,29 @@ func (r *OpenStackAnsibleEEReconciler) jobForOpenStackAnsibleEE(
...
 		addPlay(instance, h, job, hashes)
 	} else if instance.Spec.Role != nil {
 		addRoles(instance, h, job, hashes)
 	} else if len(playbook) > 0 {
 		// As we set "playbook.yaml" as default
 		// we need to ensure that Play and Role are empty before addPlaybook
+		if validate.Var(playbook, "startswith=osp.edpm.") != nil {
+			// First we check if the playbook isn't imported from collection
+			// if it is not we assume the variable holds a path.
+			valErr := validate.Var(playbook, "filepath")
```
Comment (slagle) [inline controllers/openstack_ansibleee_controller.go:373]: I don't think we can validate this. Partners will be able to build custom ansibe EE images, which may install their ansible content from packages, git, wherever. In which case, they could use their own collection name, not a filepath.

Comment (jpodivin) [reply controllers/openstack_ansibleee_controller.go:373]: That does change things a bit. I could have it match a regex of valid collection namespace and path. That way effectively covering arbitrary collections. Or, assuming that collections are accessible at runtime, get list of installed collections and verify against that. I'll look into both. Worst case scenario I can drop this check.

Comment (jpodivin) [reply controllers/openstack_ansibleee_controller.go:377]: Implemented general check with input from ansible team.

**Review by slagle — COMMENTED** [review]
+2

**Review by fao89 — APPROVED** [review]

**Comment by jpodivin** [comment]
recheck

**Pull request merged**